spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Attention.scala

```diff
@@ -15,24 +15,58 @@
  */
 package com.intel.analytics.bigdl.nn
 
-import com.intel.analytics.bigdl.Module
-import com.intel.analytics.bigdl.nn.abstractnn.TensorModule
+import javafx.scene.control.Tab
+
+import com.intel.analytics.bigdl._
+import com.intel.analytics.bigdl.nn.Graph.ModuleNode
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity, TensorModule}
 import com.intel.analytics.bigdl.tensor.Tensor
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{T, Table}
 
+import scala.language.existentials
 import scala.reflect.ClassTag
 
 /**
  * Implementation of multiheaded attention and self-attention layers.
+ *
  * @param hiddenSize hidden size
  * @param numHeads heads number
  * @param attentionDropout
  */
-class Attention[T: ClassTag](
-  val hiddenSize: Int, val numHeads: Int, val attentionDropout: Float)
- (implicit ev: TensorNumeric[T]) extends BaseModule[T] {
```

Comment:
why not extending from `BaseModule`

Reply:
If extending from `BaseModule`, `buildModel` in parent class will be called before layer initialed in child class. Then error happens.